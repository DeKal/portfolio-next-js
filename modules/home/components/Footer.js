import React from 'react'
import styled from 'styled-components/macro'
import GoTop from '~/modules/home/components/footer/GoTop'
import { Container, Row, Col } from 'react-bootstrap'
import { FaGithub } from 'react-icons/fa'

const Footer = () => (
  <FooterContainer>
    <Container>
      <Row>
        <Col md={4} xs={4}>
          <CopyRight>
            &copy;&nbsp;<Bold>Phat Ho</Bold>
          </CopyRight>
        </Col>
        <Col
          md={{ span: 4, offset: 4 }}
          xs={{ span: 4, offset: 4 }}
          className="d-flex justify-content-end"
        >
          <GitLogo href="https://github.com/DeKal/portfolio-next-js">
            <FaGithub size="40px"></FaGithub>
          </GitLogo>
        </Col>
        <GoTop />
      </Row>
    </Container>
  </FooterContainer>
)

export default Footer

const FooterContainer = styled.footer`
  position: relative;
  z-index: 100;
  width: 100%;
  background-color: ${props => props.theme.colors.colorAccent};
  color: ${props => props.theme.colors.colorLight};
  box-shadow: 0 -2px 2px -1px ${props => props.theme.colors.colorAccent};
  height: ${props => props.theme.footerHeight};

  &::before {
    content: '';
    position: absolute;
    top: -30px;
    left: 50%;
    margin-left: -40px;
    border-bottom: 40px solid ${props => props.theme.colors.colorAccent};
    border-left: 40px solid transparent;
    border-right: 40px solid transparent;
  }
`

const CopyRight = styled.p`
  margin-left: 10px;
  padding-top: 10px;
  float: left;
  line-height: 40px;

  @media screen and (max-width: 600px) {
    padding-top: 20px;
    line-height: 1;
    float: none;
    text-align: center;
  }
`

const Bold = styled.span`
  font-weight: bold;
`

const GitLogo = styled.a`
  color: ${props => props.theme.colors.colorLight};
  padding-top: 5px;
  border: none;

  &:hover {
    color: ${props => props.theme.colors.colorDark};
    text-decoration: none;
  }

  &:active {
    color: ${props => props.theme.colors.colorDark};
  }
`
