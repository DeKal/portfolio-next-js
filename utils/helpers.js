import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
