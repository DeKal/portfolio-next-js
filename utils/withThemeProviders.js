import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
