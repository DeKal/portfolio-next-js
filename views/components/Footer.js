import * as React from 'react'

const Footer = () => (
  <footer className="main-footer">
    <div className="container">
      <p className="main-footer__copyright">
        &copy;&nbsp;<span>Phat Ho</span>
      </p>
      <a className="main-footer__gotop icon icon_gotop js-anchor" href="#home">
        To top!
      </a>
    </div>
  </footer>
)

export default Footer
