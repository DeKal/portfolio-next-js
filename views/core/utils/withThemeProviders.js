import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
