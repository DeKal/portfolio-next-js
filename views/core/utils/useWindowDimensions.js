import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
