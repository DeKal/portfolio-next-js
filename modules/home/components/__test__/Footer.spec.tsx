import * as React from 'react'
import { mountWithTheme } from '~/modules/core/utils/withThemeProviders'
import Footer from '~/modules/home/components/Footer'

describe('Footer', () => {
  it('should render without throwing an error', function() {
    const wrap = mountWithTheme(<Footer />)
    expect(wrap).toMatchSnapshot()
  })
})
