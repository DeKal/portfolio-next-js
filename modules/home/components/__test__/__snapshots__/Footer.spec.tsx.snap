// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render without throwing an error 1`] = `
<Footer>
  <Footer__FooterContainer>
    <footer
      className="Footer__FooterContainer-x896os-0 iHWoeX"
    >
      <Container
        fluid={false}
      >
        <div
          className="container"
        >
          <Row
            noGutters={false}
          >
            <div
              className="row"
            >
              <Col
                md={4}
                xs={4}
              >
                <div
                  className="col-md-4 col-4"
                >
                  <Footer__CopyRight>
                    <p
                      className="Footer__CopyRight-x896os-1 dGRrGa"
                    >
                      © 
                      <Footer__Bold>
                        <span
                          className="Footer__Bold-x896os-2 iQFmgw"
                        >
                          Phat Ho
                        </span>
                      </Footer__Bold>
                    </p>
                  </Footer__CopyRight>
                </div>
              </Col>
              <Col
                className="d-flex justify-content-end"
                md={
                  Object {
                    "offset": 4,
                    "span": 4,
                  }
                }
                xs={
                  Object {
                    "offset": 4,
                    "span": 4,
                  }
                }
              >
                <div
                  className="d-flex justify-content-end col-md-4 col-4 offset-md-4 offset-4"
                >
                  <Footer__GitLogo
                    href="https://github.com/DeKal/portfolio-next-js"
                  >
                    <a
                      className="Footer__GitLogo-x896os-3 hOyXxL"
                      href="https://github.com/DeKal/portfolio-next-js"
                    >
                      <FaGithub
                        size="40px"
                      >
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 496 512",
                            }
                          }
                          size="40px"
                        >
                          <svg
                            fill="currentColor"
                            height="40px"
                            size="40px"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 496 512"
                            width="40px"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </FaGithub>
                    </a>
                  </Footer__GitLogo>
                </div>
              </Col>
              <GoTop>
                <GoTop__Container
                  href="#Home"
                >
                  <a
                    className="GoTop__Container-sc-137nslr-0 fCDaNA"
                    href="#Home"
                  >
                    <GoTop__GoTopIcon
                      name="goTop"
                    >
                      <Icon
                        className="GoTop__GoTopIcon-sc-137nslr-1 cIvAmA"
                        name="goTop"
                      >
                        <Icon__Symbol
                          className="GoTop__GoTopIcon-sc-137nslr-1 cIvAmA"
                          name="goTop"
                        >
                          <div
                            className="Icon__Symbol-sc-6mj7rl-0 bhSWal GoTop__GoTopIcon-sc-137nslr-1 cIvAmA"
                            name="goTop"
                          />
                        </Icon__Symbol>
                      </Icon>
                    </GoTop__GoTopIcon>
                  </a>
                </GoTop__Container>
              </GoTop>
            </div>
          </Row>
        </div>
      </Container>
    </footer>
  </Footer__FooterContainer>
</Footer>
`;
